pkg/distributor/distributor.go

```diff
@@ -75,10 +79,10 @@ func (cfg *Config) RegisterFlags(fs *flag.FlagSet) {
 	fs.BoolVar(&cfg.ShardStreams.LoggingEnabled, "distributor.stream-sharding.logging-enabled", false, "Enable logging when sharding streams")
 }
 
-// StreamSharder manages the state necessary to shard streams.
-type StreamSharder interface {
-	ShardCountFor(stream logproto.Stream) (int, bool)
-	IncreaseShardsFor(stream logproto.Stream)
+// RateStore manages the ingestion rate of streams, populated by data fetched from ingesters.
+type RateStore interface {
+	RateFor(stream logproto.Stream) (int, error)
+	StoreRate(stream logproto.Stream, rate int) error
```

Comment:
We won't need this on the interface. The rate store will take total care of how sharding should work

Reply:
agreed, good catch.